### ISSUE TITLE
Is there some parameters that I can run it concurrently？

### ISSUE THREAD

**Issue by forrwill** [EichlerLab/smrtsv2#1]
I was running the smrtsv2, and I found that it is one-by-one running with snakemake on sge. it is too slowly. Is there anyone can tell me that is there something is wrong?

**Comment by paudano** [comment]
SMRT-SV is currently designed to distribute over an SGE cluster using DRMAA (https://pypi.org/project/drmaa/). `smrtsv.py` passes several parameters through to Snakemake. If you can distribute a Snakemake pipeline through DRMAA, then you can set the same parameters to distribute SMRT-SV.

Key parameters:
  * --distribute
    * Tells SMRT-SV to submit jobs and distribute them.
  * --cluster-params <param string>
    * A string specific to your cluster. It should have "{{cluster.cpu}}", "{{cluster.mem}}", "{{cluster.rt}}", and "{{cluster.params}}". The values for those placeholders are read from the cluster configuration file (see --cluster-config). This string **MUST** begin with a space.
  * --cluster-config <config json>
    * Path to the cluster configuration file. This tells SGE what the resource requirements are for each rule. In the Eichler lab, we use "cluster.eichler.json". You may need to customize this file for your cluster, but use it as a guide.
  * --drmaalib </path/to/libdrmaa.so.1.0>
    * If `DRMAA_LIBRARY_PATH` is not set in the environment, then this option is required. It is the full path to `libdrmaa.so.1.0`.
  * --jobs <n>
    * Number of jobs to run in the current stage (align, detect, assemble, or call). The `run` command ignores this parameter.
  * --runjobs <align,detect,assemble,call>
    * Sets the number of jobs for each stage in the order listed above. For example, "25,20,120,10" would run 25 alignment jobs, 20 detect jobs, 120 assemblies, and 10 call jobs concurrently.

Note that all these parameters should appear on the command line before the SMRT-SV command to be run. The only exception is `--runjobs`, which must appear after `run` (see below).

For example:
```
SMRTSV_DIR=/path/to/smrtsv2

${SMRTSV_DIR}/smrtsv.py \
  --cluster-config ${SMRTSV_DIR}/cluster.eichler.json \
  --drmaalib /usr/lib/libdrmaa.so.1.0 \
  --distribute \
  run \
  --runjobs "25,20,120,10" \
  /path/to/reference.fasta \
  /path/to/reads.fofn
```

Most of these options are passed directly to Snakemake, so you can look at Snakemake arguments for more information. If it's useful, `smrtsvlib/args.py` has the options and their default values, and `run_snake_target()` in `smrtsvlib/smrtsvrunner.py` builds the command-line options and feeds them to Snakemake.

**Comment by forrwill** [comment]
thanks  for your reply and I really appreciate it!
then, I have run the pipline on the sge, but it can only submit one job  automately. can I submit jobs on the sge concurrently?
Thank you!

**Comment by paudano** [comment]
This is what `--jobs` and `--runjobs` does, it tell the pipeline how many concurrent jobs to submit. If you set `--jobs 100` and there are 1,000 jobs ready to run, it will submit 100 of them. When one finishes, the next job in the available pool is run.

**Comment by forrwill** [comment]
> This is what `--jobs` and `--runjobs` does, it tell the pipeline how many concurrent jobs to submit. If you set `--jobs 100` and there are 1,000 jobs ready to run, it will submit 100 of them. When one finishes, the next job in the available pool is run.

Yes，it is! thank you

**Comment by forrwill** [comment]
I met a question when in the assembly.and It give the bug "variantCaller: command not found",but I can not find it. would  you please give me some guides about the question？

**Comment by paudano** [comment]
variantCaller is the program that runs arrow to polish assemblies. It is part of the "GenomicConsensus" PacBio package. If you don't have it installed somewhere, I would install it through BioConda and place the path to the executable it installs in PATH for SMRT-SV (https://github.com/PacificBiosciences/pbbioconda).

**Comment by forrwill** [comment]
Thank you! but I am really sorry that I can not find the tool of " alignfixup". the log file tell me that "/usr/bin/bash: alignfixup: command not found"

**Comment by paudano** [comment]
`alignfixup` was part of the `dist` directory, and I need to pull it back in.

I am about to redo the build system so that SMRT-SV can set up all its dependencies before building, so this should be less painful within a few weeks.

I will let you know when I have an update.

**Comment by forrwill** [comment]
thank you!
could you give me some suggestions when using the Illumina data? such as which software results is more accurate？

**Comment by paudano** [comment]
I just pushed an update with a build system for dependencies. If you CD into "dep" and run "make", it should build everything, including "alignfixup", and place it in "dep/bin". SMRT-SV will search "dep/bin" before any other locations, so if you are using your own installed tools, SMRT-SV will start using the tools installed in "dep" instead (e.g. snakemake, canu, blasr, samtools, etc). I just added this and have not tested it yet.

For short reads, the most current resource I know of is the HGSVC paper:
https://www.biorxiv.org/content/early/2018/06/13/193144

Several short-read callers were run and compared in that study. I know there are other papers, but these are not tools I often run.

Let me know if it gets past alignfixup or if you have other problems.
